### PR TITLE
Chapter15

### PR DESCRIPTION
**Pull request by danieljharris**

### CODE REVIEW

**Comment by velimir** [comment]
there're changes from chapter12 and chapter10... so I guess you'll have conflicts. A good rule of thumb - do not include to pull request (PR) anything that is not related to PR subject.

File: Chapter15/proxy.erl [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    {error, closed} ->
+      ok
+  end.
+
+% the http module is no longer used and I could not get the newer httpc module
+% to return the site in the same format as the http module would have 
+
+handle([<<_Get:4/binary, Tail/binary>>], _Count, Socket) ->
+  Url = extract_url(Tail, []),
+  % {ok, Site} = httpc:request(Url),
```
Comment (velimir) [inline Chapter15/proxy.erl]: shouldn't `httpc:request/1` do it?
Something like:
```erlang
{ok, {{_HttpVersion, 200, "OK"}, _Headers, _Body}} = httpc:request("http://erlang.org" ++ Url).
```

File: Chapter15/peer.erl [hunk 2]
```diff
@@ -0,0 +1,86 @@
+-module(peer).
+-export([start/0, connect/1, send/1, stop/0]).
+-export([wait_connect/1]).
+-export([loop/1]).
+
+% Exercise 15-3: Peer to Peer
+% I couldent figure out how to get the crypto module working for encryption
```
Comment (velimir) [inline Chapter15/peer.erl]: there's docs about crypto: https://github.com/joearms/crypto_tutorial

File: Chapter15/peer.erl [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    undefined ->
+      register(peer, spawn(?MODULE, loop, [[]])),
+      peer ! start;
+    _Other ->
+      {error, already_started}
+  end.
+
+connect(IpAddress) ->
+  case catch(peer ! {connect, IpAddress}) of
+    {'EXIT', _} -> {not_connected};
```
Comment (velimir) [inline Chapter15/peer.erl]: shouldn't it be `{error, not_started}`?
Actually, exercise has a defined interface for these functions. Can you please fix them according to specifications?

**Review by velimir — CHANGES_REQUESTED** [review]
LGTM, except some comments

File: Chapter15/proxy.erl (hunk 1 above)
Comment (danieljharris) [reply Chapter15/proxy.erl]: All fixed

File: Chapter15/peer.erl (hunk 2 above)
Comment (danieljharris) [reply Chapter15/peer.erl]: I figured out how to implement simple encryption using hardcoded keys, if you would like me to implement a better solution (not using hardcoded keys) let me know and I'll get it updated.

File: Chapter15/peer.erl (hunk 3 above)
Comment (danieljharris) [reply Chapter15/peer.erl]: Added the missing "error" atoms

**Review by velimir — APPROVED** [review]

**Pull request merged**